ml-agents-envs/mlagents_envs/environment.py

```diff
@@ -147,55 +147,64 @@ def __init__(
-        logger.debug("The true file name is {}".format(true_filename))
         launch_string = None
+        true_filename = os.path.basename(os.path.normpath(env_path))
         if platform == "linux" or platform == "linux2":
-            candidates = glob.glob(os.path.join(cwd, file_name) + ".x86_64")
+            candidates = glob.glob(os.path.join(cwd, env_path) + ".x86_64")
             if len(candidates) == 0:
-                candidates = glob.glob(os.path.join(cwd, file_name) + ".x86")
+                candidates = glob.glob(os.path.join(cwd, env_path) + ".x86")
             if len(candidates) == 0:
-                candidates = glob.glob(file_name + ".x86_64")
+                candidates = glob.glob(env_path + ".x86_64")
             if len(candidates) == 0:
-                candidates = glob.glob(file_name + ".x86")
+                candidates = glob.glob(env_path + ".x86")
             if len(candidates) > 0:
                 launch_string = candidates[0]
 
         elif platform == "darwin":
             candidates = glob.glob(
-                os.path.join(
-                    cwd, file_name + ".app", "Contents", "MacOS", true_filename
-                )
+                os.path.join(cwd, env_path + ".app", "Contents", "MacOS", true_filename)
             )
             if len(candidates) == 0:
                 candidates = glob.glob(
-                    os.path.join(file_name + ".app", "Contents", "MacOS", true_filename)
+                    os.path.join(env_path + ".app", "Contents", "MacOS", true_filename)
                 )
             if len(candidates) == 0:
                 candidates = glob.glob(
-                    os.path.join(cwd, file_name + ".app", "Contents", "MacOS", "*")
+                    os.path.join(cwd, env_path + ".app", "Contents", "MacOS", "*")
                 )
             if len(candidates) == 0:
                 candidates = glob.glob(
-                    os.path.join(file_name + ".app", "Contents", "MacOS", "*")
+                    os.path.join(env_path + ".app", "Contents", "MacOS", "*")
                 )
             if len(candidates) > 0:
                 launch_string = candidates[0]
         elif platform == "win32":
-            candidates = glob.glob(os.path.join(cwd, file_name + ".exe"))
+            candidates = glob.glob(os.path.join(cwd, env_path + ".exe"))
             if len(candidates) == 0:
-                candidates = glob.glob(file_name + ".exe")
+                candidates = glob.glob(env_path + ".exe")
             if len(candidates) > 0:
                 launch_string = candidates[0]
+        return launch_string
+
+    def executable_launcher(self, file_name, docker_training, no_graphics, args):
+        file_name = (
+            file_name.strip()
+            .replace(".app", "")
+            .replace(".exe", "")
+            .replace(".x86_64", "")
+            .replace(".x86", "")
+        )
+        true_filename = os.path.basename(os.path.normpath(file_name))
```

Comment:
Could you move this line and the one above into `environment_launch_check`?  I'm not 100% sure it's necessary, but seems like it is more of the logic around validating the path.